qos/evm/preset_responses.go

```diff
@@ -0,0 +1,27 @@
+package evm
+
+import (
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// errResponseEmptyEndpointResponse represents a JSON-RPC error response for when
+// an endpoint returns an empty response body.
+// This indicates server misbehavior, triggers removal of the endpoint from the selection pool,
+// and tells the client they can retry with a different endpoint.
+var errResponseEmptyEndpointResponse = jsonrpc.GetErrorResponse(
+	jsonrpc.ID{}, // Use request's original ID if present
+	-32000,       // JSON-RPC server error code
+	"Server error: Received an empty response. Server will be dropped from the selection pool. Please try again.", // Error Response Message
```

Comment:
Wdyt of this instead? Server has a few too many meanings here.

```suggestion
	"Endpoint (data/service node error): Received an empty response. The endpoint will be dropped from the selection pool. Please try again.", // Error Response Message
```